zerver/lib/upload.py

```diff
@@ -835,10 +844,13 @@ def upload_message_file(
         user_profile: UserProfile,
         target_realm: Optional[Realm] = None,
     ) -> str:
+        if target_realm is None:
+            target_realm = user_profile.realm
+
         path = self.generate_message_upload_path(str(user_profile.realm_id), uploaded_file_name)
```

Comment:
This should be `target_realm.id`, right?

Reply:
Yeah, good catch. Fixed. The other backend already uses target_realm, so that one's fine.